src/git-command-manager.ts

```diff
@@ -0,0 +1,399 @@
+import * as core from '@actions/core'
```

Comment:
ideally we would use jest or some kind of testing framework to write unit tests around these classes, rather than just the shell scripts. That would help other people write more tests in the future and make the tests more bite sized to the JS classes, plus define expected input/output

Reply:
+1 there is a couple places in this module with logic that would be interesting to unit test

most of the logic is uninteresting though. biggest value for investment simple e2e w/ a couple shell scripts to assert expected layout

Reply:
chasing down other details right now, will try to circle back and add jest

Reply:
I added jest and refactored some of the interesting logic out into separate modules so i could isolate them via unit tests.

note, i intentionally did not invest in jest tests that make heavy use of mocking (high investment cost, low return). Rather I pulled interesting logic out into testable modules. Otherwise i'm leaning on the E2E coverage (easy to write, doesnt need to cover every scenario but pretty decent coverage with just a few e2e tests)